use statically known true instead of guard size oblivious in bmm and mm inductor decompositions .

laithsakka opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #148893

this was discussed with @eellison and he recommended using  statically_known_true here, the intuition is. We already have 0/1 specializations in place, if we reach those checks with dynamic shapes that are not already specialized 
then we do not want them to specialize them, "a recompilation here is not justified".
Those are all non-semantic changing optimizations.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @chenyang78 @kadeng @muchulee8 @amjames @chauhang @aakhundov

laithsakka committed:
use statically known true instead of guard size oblivous

[ghstack-poisoned]

laithsakka committed:
Update on "[DRAFT] use statically known true instead of guard size oblivous"




[ghstack-poisoned]

eellison reviewed (changes requested):
request review when tests are fixed

laithsakka committed:
Update on "[DRAFT] use statically known true instead of guard size oblivous"




[ghstack-poisoned]

laithsakka committed:
Update on "use statically known true instead of guard size oblivious in bmm and mm decompositions . "


this was discussed with eellison and he recommended using  statically_known_true here, the intuition is. We already have 0/1 specializations in place, if we reach those checks with dynamic shapes that are not already specialized 
then we do not want them to specialize them, "a recompilation here is not justified".
Those are all non-semantic changing optimizations.

cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy chenyang78 kadeng muchulee8 amjames chauhang aakhundov

[ghstack-poisoned]

eellison reviewed (approved):
I'm a little worried this will regress performance in gpt-fast, but, I agree that today this is not the best experience. 

Getting these firing will be a nice end result of @bobrenjc93's work

laithsakka commented:
We can always go back to guard_or_false if this end up being a regression

laithsakka commented:
@pytorchbot merge

laithsakka commented:
@pytorchbot merge -f "GH is broken"
